lib/wibox/widget/slider.lua

```diff
@@ -0,0 +1,548 @@
+        else
+            x_offset, y_offset = margins.left or 0, margins.top or 0
+            handle_width  = handle_width  -
+                (margins.left or 0) - (margins.right  or 0)
+            handle_height = handle_height -
+                (margins.top  or 0) - (margins.bottom or 0)
+        end
+    end
+
+    local value = self._private.value or self._private.min or 0
+
+    -- Get the widget size back to it's non-transfored value
+    --local matrix = gmatrix.from_cairo_matrix(cr:get_matrix())
+    local min, _, interval = get_extremums(self)
+    local rel_value = ((value-min)/interval) * (width-handle_width)
+
+    cr:translate(x_offset + rel_value, y_offset)
+
+    local handle_border_width = self._private.handle_border_width
+        or beautiful.slider_handle_border_width
+        or properties.handle_border_width or 0
+
+    handle_shape(cr, handle_width, handle_height)
+
+    if handle_border_width > 0 then
+        cr:fill_preserve()
+    else
+        cr:fill()
+    end
+
+    -- Draw the handle border
+    if handle_border_width > 0 then
+        local handle_border_color = self._private.handle_border_color
+            or beautiful.slider_handle_border_color
+            or properties.handle_border_color
+
+        if handle_border_color then
+            cr:set_source(color(handle_border_color))
+        end
+
+        cr:set_line_width(handle_border_width)
+        cr:stroke()
+    end
+end
+
+function slider:fit(_, width, height)
+    -- Use all the space, this should be used with a constraint widget
+    return width, height
+end
+
+local event = {}
+
+function event.mouse_enter(self)
+    self:emit_signal("widget::redraw_needed") --TODO
+    local cursor = self._private.cursor or "fleur"
+    capi.root.cursor(cursor)
+end
+
+function event.mouse_leave(self)
+    self:emit_signal("widget::redraw_needed") --TODO
+--     capi.root.cursor("fleur")
+end
+
+local rect_order = {"x", "y", "width", "height"}
+
+-- Move the handle to the correct location
+local function move_handle(self, x, y, geo)
+    -- Fix the X position when used in a rotate or mirror widget
+
+    -- Remove the translation to bring {x, y} and the rect on the same origin
+    local rem_translate = geo.matrix_to_device*gmatrix.identity
+    rem_translate.x0, rem_translate.y0 = 0, 0
+
+    local rect = {rem_translate:transform_rectangle(
+        0,
+        0,
+        geo.width,
+        geo.height
+    )}
+
+    -- Floating point precision is here both too much and not enough
+    for k, v in ipairs(rect) do
+        rect[rect_order[k]] = util.round(v)
+    end
+
+    local px = rem_translate:transform_point(x, y)
+
+    -- If the transformation moved the rectangle away from {0,0}, compensate
+    if rect.x < 0 then
+        px = rect.width + px
+    end
+
+    --HACK Cairo 0,0 is top left, a matrix is bottom left, funny...
+    if util.round(rem_translate:transform_point(0, 100)) ~= 0 then
+        px = -(px - rect.width)
+    end
```

Comment:
I don't get the math here and it doesn't seem to me like the author really understood it either. A matrix doesn't have a "bottom left"...?!?